uPortal-core/src/main/java/org/apereo/portal/portlet/om/PortletLifecycleState.java

```diff
@@ -15,21 +15,34 @@
 package org.apereo.portal.portlet.om;
 
 public enum PortletLifecycleState {
+
     CREATED(0),
     APPROVED(1),
     PUBLISHED(2),
     EXPIRED(3),
     MAINTENANCE(4);
 
-    public static final String MAINTENANCE_MODE_PARAMETER_NAME =
-            "PortletLifecycleState.inMaintenanceMode";
-
     private int order;
 
-    private PortletLifecycleState(int order) {
+    PortletLifecycleState(int order) {
         this.order = order;
     }
 
+    public static PortletLifecycleState forOrderValue(int orderValue) {
+        PortletLifecycleState rslt = null;
+        for (PortletLifecycleState state : PortletLifecycleState.values()) {
+            if (state.getOrder() == orderValue) {
+                rslt = state;
+                break;
+            }
+        }
+        if (rslt == null) {
```

Comment:
BTW, I think this is excellent feedback.  I love having an extra pair of eyes to help catch the "boneheaded" moments.

In this case, we're good.  Certainly -- if `state` were null on line 34 we'd get an NPE.

But we are checking `rslt` for null on this line (#39).  We set `rslt` to null when we defined it on line 32.  We are checking if it is _still_ null, which would indicate that we don't have an instance of `PortletLifecycleState` that matches the specified `orderValue`.